state/state.go

```diff
@@ -1316,6 +1316,15 @@ func (st *State) AddApplication(args AddApplicationArgs) (_ *Application, err er
 			ops = append(ops, resOps...)
 		}
 
+		isSidecar, err := app.IsSidecar()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if isSidecar {
+			if err := resetSequence(st, app.Tag().String()); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
```

Comment:
This needs to be done as ops via the txn runner, otherwise if the add application fails,the sequence would still be reset.

Reply:
Yeah, agreed. changed to insert a txn.Op into current ops.

Reply:
We need to reset the sequence before addUnitOpsWithCons.
Because addUnitOpsWithCons will fetch the sequence to decide the unit name.